pkg/machine/compression/compression_test.go

```diff
@@ -89,3 +95,37 @@ func TestImageCompression_String(t *testing.T) {
 		})
 	}
 }
+
+func Test_Decompress(t *testing.T) {
+	type args struct {
+		src string
+		dst string
+	}
+
+	type want string
+
+	tests := []struct {
+		name string
+		args args
+		want want
+	}{
+		{name: "zip", args: args{src: "./testfiles/sample.zip", dst: "./testfiles/hellozip"}, want: "zip\n"},
+		{name: "xz", args: args{src: "./testfiles/sample.xz", dst: "./testfiles/helloxz"}, want: "xz\n"},
+		{name: "gzip", args: args{src: "./testfiles/sample.gz", dst: "./testfiles/hellogz"}, want: "gzip\n"},
+		{name: "bzip2", args: args{src: "./testfiles/sample.bz2", dst: "./testfiles/hellobz2"}, want: "bzip2\n"},
+		{name: "zstd", args: args{src: "./testfiles/sample.zst", dst: "./testfiles/hellozstd"}, want: "zstd\n"},
+		{name: "uncompressed", args: args{src: "./testfiles/sample.uncompressed", dst: "./testfiles/hellozuncompressed"}, want: "uncompressed\n"},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			srcVMFile := &define.VMFile{Path: tt.args.src}
+			dstFilePath := tt.args.dst
+			defer os.Remove(dstFilePath)
+			err := Decompress(srcVMFile, dstFilePath)
+			assert.NoError(t, err)
+			data, err := os.ReadFile(dstFilePath)
+			assert.NoError(t, err)
```

Comment:
(`require.NoError` in the two cases above, when it makes no sense to continue the test. (Cleaning this up can wait only if the test ever fails…). )

Reply:
updated